pyproject.toml

```diff
@@ -36,29 +36,49 @@ License = "https://github.com/rapidsai/legate-raft/blob/main/LICENSE"
 
 [project.optional-dependencies]
 test = [
-    "cupynumeric==24.11.*",
+    "cupynumeric==25.01.*,>=0.0.0.dev0",
     "hypothesis>=6",
     "joblib",
     "pytest>=7.0",
     "scikit-learn",
 ] # This list was generated by `rapids-dependency-file-generator`. To make changes, edit dependencies.yaml and run `rapids-dependency-file-generator`.
 
 [build-system]
+build-backend = "rapids_build_backend.build"
+requires = [
+    "rapids-build-backend>=0.3.2,<0.4.0.dev0",
+    "scikit-build-core[pyproject]>=0.10.0",
+] # This list was generated by `rapids-dependency-file-generator`. To make changes, edit dependencies.yaml and run `rapids-dependency-file-generator`.
+
+[tool.rapids-build-backend]
+build-backend = "scikit_build_core.build"
+dependencies-file = "dependencies.yaml"
+matrix-entry = "cuda_suffixed=true"
 requires = [
     "cmake>=3.26.4,!=3.30.0",
-    "cython>=3.0.0",
-    "legate-dataframe==24.11.*",
-    "legate==24.11.01",
-    "libraft==24.08.*,>=0.0.0a0",
-    "librmm==24.08.*,>=0.0.0a0",
+    "cython>=3.0.3",
+    "legate-dataframe==25.01.*,>=0.0.0.dev0",
+    "legate==25.01.*,>=0.0.0.dev0",
+    "libraft==24.10.*,>=0.0.0a0",
+    "librmm==24.10.*,>=0.0.0a0",
     "ninja",
-    "scikit-build>=0.18.0",
-    "setuptools>=70.0",
 ] # This list was generated by `rapids-dependency-file-generator`. To make changes, edit dependencies.yaml and run `rapids-dependency-file-generator`.
-build-backend = "setuptools.build_meta"
 
-[tool.setuptools.dynamic]
-version = {file = "legate_raft/VERSION"}
+[tool.scikit-build.metadata.version]
+provider = "scikit_build_core.metadata.regex"
+input = "./VERSION"
+regex = "(?P<value>.*)"
+
+[tool.scikit-build]
+build-dir = "build/{wheel_tag}"
+cmake.build-type = "Release"
+cmake.version = "CMakeLists.txt"
+minimum-version = "build-system.requires"
+ninja.make-fallback = true
+sdist.exclude = ["*tests*"]
+sdist.include = ["legate_raft/install_info.py"]
```

Comment:
`scikit-build-core` honors `.gitignore`, took me a while to realize why this was missing.

Reply:
oh wow! I actually did not know that, thanks for the comment.